### PR TITLE
Adding the files needed to allow the devs to mount their local code

### PR DESCRIPTION
**Pull request by danakim**
Hi @quartje ! This PR adds the possibility to mount local directories with code into one of the services that the user chooses.

It uses:
* `docker-compose.yml` files that override settings for just the service in question. Each service directory has one of these files. They mount a local volume over `/var/html/www` in the container
* the local directory to mount is chosen from an environment variable because docker compose files have the ability to read env vars and replace them in the file, like a templating system
* a startup script called `start-dev-env.sh` that takes two parameters: the service name and the local directory to mount. Example: `start-dev-env.sh webauthn /home/dan/Stepup-webauthn` (the recommended way would be to use absolute paths).
* the startup script uses these two parameters to read the docker compose override file from the service's directory and replace the code path in that file (by reading it as an env var)

NOTES:
* if you do not like the idea of a startup script (it is a really simple one), we can just instruct the devs on how to run this manually:

    `export CODE_PATH=<path_to_code>`
    `docker-compose -f docker-compose.yml -f <service_directory>/docker-compose.override.yml up`

* I also had another idea where we can have a single docker-compose.override.yml file in the root directory that we can use as a template. And use some sed / bash magic to replace what we need in that file for any service required. But I think it is more prone to errors and also having an override file for each service allows us in the future to do more dev customisation for each service.

### CODE REVIEW

File: stepup/start-dev-env.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env bash
+
+# Read the command line parameters to configure the dev env
+export SERVICE=$1
+export CODE_PATH=$2
+
+# Use docker compose to start the environment but with the modified override file
+docker-compose -f docker-compose.yml -f ./${SERVICE}/docker-compose.override.yml up
```
Comment (quartje) [inline stepup/start-dev-env.sh:8]: Nice and clean, very understandable. Two things could be added:

1) Printing the actual command that is run. Very usefull for debugging, and also for the devs to see what's happening
2) A way to add all parameters that you'd add as parameters to the script. You can eg add -d to the docker-compose command, or other command line options:

So: start-dev-env.sh middleware ../middleware -d would do:
docker-compose -f docker-compose.yml -f ../middleware/docker-compose-override.yml up -d

File: stepup/gateway/docker-compose.override.yml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+services:
```
[the remark this answers is not in the document]
Comment (danakim) [reply stepup/gateway/docker-compose.override.yml:1]: @quartje - Ah yes! I forgot about this. Thanks for reminding me. I have committed now to this PR / branch the changes to use the dev image.

File: stepup/start-dev-env.sh (hunk 1 above)
Comment (danakim) [reply stepup/start-dev-env.sh]: Thanks @quartje ! Nice ideas ;) I have implemented both them in my latest commit to this branch / PR.

**Comment by danakim** [comment]
@quartje - I have resolved both your requests to this PR. When you have the time, please test.

**Pull request merged**